### ISSUE TITLE
forloop.first does not work

### ISSUE THREAD

**Issue by Burstaholic** [simonw/djangode#8]
in a {% for %} template tag, forloop.first is always false. In a related issue, forloop.counter and forloop.counter0 always show you starting in the second iteration.

**Comment by simonw** [comment]
I'm no longer actively maintaining my branch of djangode, but https://github.com/whitelynx/djangode is an actively maintained branch that has a fix for this problem.

**Comment by Burstaholic** [comment]
Heh, I'm actually working with him, and it's turning out to be quite a
difficult fix - the naive fix we tried first broke pretty much every
iteration construct (thus we had to revert it), and we're trying to figure
out where the root of the issue lies.

If you are able to give us any insight into why this bit of code behaves
the way it does it would be very helpful.

https://github.com/whitelynx/djangode/commit/2679e4caf1eafe5178d50ee14789e675c17a8a12

If that's not practical for you I understand, and I do appreciate your
taking the time to respond to me.

Thanks,

Travis
